qemu/tests/nic_hotplug.py

```diff
@@ -109,6 +113,23 @@ def device_add_nic(pci_model, netdev, device_id):
         add_output = vm.monitor.send_args_cmd(pci_add_cmd)
         return add_output
 
+    def reboot_vm(vm, need_session=True, serial=True):
+        """
+        Optimize the console usage while doing guest reboot. and check the
+        guest alive status
+        :vm : the vm object to be rebooted
+        :param need_session: a flag to mark if we need to give a session param
+                             to 'reboot' function
+        :param serial: determine if we will try to reboot guest through serial
+                       port
+        """
+        session = None
+        if need_session:
+            session = vm.wait_for_serial_login(timeout=login_timeout)
```

Comment:
there is session in reboot,
why need to login here again
        def _shell_reboot(session, timeout):
            if not session:
                if not serial:
                    session = self.wait_for_login(nic_index=nic_index,
                                                  timeout=timeout)
                else:
                    session = self.wait_for_serial_login(timeout=timeout)
            reboot_cmd = self.params.get("reboot_command")

Reply:
Updated thanks